Tests/UnitTests/Core/Detector/CylindricalDetectorFromBlueprintTests.cpp

```diff
@@ -0,0 +1,251 @@
+      100, 105., detectorHz - 2 * pixelEcHz};
+  auto pixelBarrelL1 = std::make_unique<Acts::Experimental::Blueprint::Node>(
+      "pixel_barrel_l1", Acts::Transform3::Identity(),
+      Acts::VolumeBounds::eCylinder, pixelBarrelL1Boundaries,
+      pixelBarrelL1Structure);
+  pixelBarrel->add(std::move(pixelBarrelL0));
+  pixelBarrel->add(std::move(pixelBarrelL1));
+
+  Acts::Transform3 pixelPecTransform =
+      Acts::Transform3::Identity() *
+      Acts::Translation3(0., 0., detectorHz - pixelEcHz);
+
+  auto pixelPec = std::make_unique<Acts::Experimental::Blueprint::Node>(
+      "pixel_pec", pixelPecTransform, Acts::VolumeBounds::eCylinder,
+      pixelEcBoundaries, pixelEcBinning);
+
+  std::vector<Acts::ActsScalar> pixelPecBoundaries = {pixelIr + 2, pixelOr - 7.,
+                                                      10.};
+
+  auto pixelPecLayerStructure =
+      std::make_shared<SurfaceBuilder<Acts::DiscSurface>>(
+          pixelPecTransform, pixelIr + 10., pixelOr - 10.);
+
+  auto pixelPecLayer = std::make_unique<Acts::Experimental::Blueprint::Node>(
+      "pixel_pec_layer", pixelPecTransform, Acts::VolumeBounds::eCylinder,
+      pixelPecBoundaries, pixelPecLayerStructure);
+
+  pixelPec->add(std::move(pixelPecLayer));
+
+  // Adding pixel
+  pixel->add(std::move(pixelNec));
+  pixel->add(std::move(pixelPec));
+  pixel->add(std::move(pixelBarrel));
+
+  detectorBpr->add(std::move(pixel));
+
+  // An Indexed volume finder will be attached
+  std::vector<Acts::BinningValue> rootVolumeBinning = {Acts::binZ, Acts::binR};
+  detectorBpr->rootVolumeFinderBuilder =
+      std::make_shared<Acts::Experimental::IndexedRootVolumeFinderBuilder>(
+          rootVolumeBinning);
+
+  // A geo ID generator
+  detectorBpr->geoIdGenerator =
+      std::make_shared<Acts::Experimental::GeometryIdGenerator>(
+          Acts::Experimental::GeometryIdGenerator::Config{},
+          Acts::getDefaultLogger("RecursiveIdGenerator",
+                                 Acts::Logging::VERBOSE));
+
+  // Complete and fill gaps
+  Acts::Experimental::detail::BlueprintHelper::fillGaps(*detectorBpr);
+
+  std::fstream fs("cylindrical_detector_blueprint.dot", std::ios::out);
+  detectorBpr->dotStream(fs, "blueprint2cylinder");
+  fs.close();
+
+  // ----------------------------- end of blueprint
+
+  // Create a Cylindrical detector builder from this blueprint
+  auto detectorBuilder =
+      Acts::Experimental::CylindricalContainerBuilder::createFromBlueprint(
+          *detectorBpr, Acts::Logging::VERBOSE);
+
+  // Detector builder
+  Acts::Experimental::DetectorBuilder::Config dCfg;
+  dCfg.auxiliary =
+      "*** Test : auto generated cylindrical detector builder  ***";
+  dCfg.name = "Cylindrical detector from blueprint";
+  dCfg.builder = detectorBuilder;
+  dCfg.geoIdGenerator = detectorBpr->geoIdGenerator;
+
+  auto detector = Acts::Experimental::DetectorBuilder(dCfg).construct(tContext);
+
+  BOOST_TEST(detector != nullptr);
```

Comment:
I think we moved away from using the `BOOST_TEST` macro.

Reply:
Correct - mea culpa - removed.